buildSrc/src/main/groovy/airbyte-python.gradle

```diff
@@ -8,6 +10,46 @@ class AirbytePythonConfiguration {
     String moduleDirectory
 }
 
+class Helpers {
+    static addTestTaskIfTestFilesFound(Project project, String testFilesDirectory, String taskName, taskDependencies) {
+        """
+        This method verifies if there are test files in a directory before adding the pytest task to run tests on that directory. This is needed 
+        because if there are no tests in that dir and we run pytest on it, it exits with exit code 5 which gradle takes to mean that the process
+        failed, since it's non-zero. This means that if a module doesn't need a unit or integration test, it still needs to add a dummy test file
+        like: 
+        
+        ```
+        def make_ci_pass_test():
+            assert True
+        ```
+        
+        So we use this method to leverage pytest's test discovery rules (https://docs.pytest.org/en/6.2.x/goodpractices.html#conventions-for-python-test-discovery)
+        to selectively run pytest based on whether there seem to be test files in that directory. 
+       
+       
+        See https://github.com/airbytehq/airbyte/issues/4979 for original context
+        """
+        if (project.file(testFilesDirectory).exists()) {
+
+            project.projectDir.toPath().resolve(testFilesDirectory).traverse(type: FileType.FILES, nameFilter: ~/(^test_.*|.*_test)\.py$/) { file ->
+                project.task(taskName, type: PythonTask, dependsOn: taskDependencies) {
```

Comment:
what happens here if there are multiple files found? will this add a task for each file?

Reply:
the `return TERMINATE` ensures that traversal stops if a file is found, and so this is called at most once. Added as a comment.

Reply:
thanks!